src/TestProblem/Hydro/CCSN/Flag_CCSN.cpp

```diff
@@ -0,0 +1,64 @@
+#include "GAMER.h"
+
+
+extern double CCSN_MaxRefine_RadFac;
+
+
+
+//-------------------------------------------------------------------------------------------------------
+// Function    :  Flag_Lightbulb
+// Description :  Check if the element (i,j,k) of the input data satisfies the user-defined flag criteria
+//
+// Note        :  1. Invoked by "Flag_Check" using the function pointer "Flag_User_Ptr"
+//                   --> The function pointer may be reset by various test problem initializers, in which case
+//                       this function will become useless
+//                2. Enabled by the runtime option "OPT__FLAG_USER"
+//                3. For lightbulb test problem
+//
+// Parameter   :  i,j,k       : Indices of the target element in the patch ptr[ amr->FluSg[lv] ][lv][PID]
+//                lv          : Refinement level of the target patch
+//                PID         : ID of the target patch
+//                Threshold   : User-provided threshold for the flag operation, which is loaded from the
+//                              file "Input__Flag_User"
+//                              In order of radius_min, radius_max, threshold_dens
+//
+// Return      :  "true"  if the flag criteria are satisfied
+//                "false" if the flag criteria are not satisfied
+//-------------------------------------------------------------------------------------------------------
+bool Flag_Lightbulb( const int i, const int j, const int k, const int lv, const int PID, const double *Threshold )
+{
+
+   bool Flag = false;
+
+   const double dh        = amr->dh[lv];
+   const double Center[3] = { amr->BoxCenter[0], amr->BoxCenter[1], amr->BoxCenter[2] };
+   const double Pos   [3] = { amr->patch[0][lv][PID]->EdgeL[0] + (i+0.5)*dh,
+                              amr->patch[0][lv][PID]->EdgeL[1] + (j+0.5)*dh,
+                              amr->patch[0][lv][PID]->EdgeL[2] + (k+0.5)*dh  };
+
+   const double dx = Center[0] - Pos[0];
+   const double dy = Center[1] - Pos[1];
+   const double dz = Center[2] - Pos[2];
+   const double r  = sqrt(  SQR( dx ) + SQR( dy ) + SQR( dz )  );
+
+   const real (*Rho )[PS1][PS1] = amr->patch[ amr->FluSg[lv] ][lv][PID]->fluid[DENS];
+
+
+// TODO: fine-tune the criteria
+// always allow the highest level can be reached in the region with r < 30 km
+   if ( r * UNIT_L < 3e6 )
+   {
+      Flag = true;
+   }
+
+   else
+   {
+//    if density is larger than the threshold (equivalent to Input__Flag_Rho)
+//    and the cell width at lv+1 is larger than the threshold `r * CCSN_MaxRefine_RadFac` (equivalent to Flag_Region)
+      if (  ( Rho[k][j][i] > Threshold[0] )  &&  ( 0.5 * dh > r * CCSN_MaxRefine_RadFac )  )   Flag = true;
```

Comment:
- I update the `README` to state these threshold values must be in the code units. BTW, I am thinking if we need a boolean runtime parameter like `OPT__FLAG_UNIT` that automatically converts the values in refinement tables to the code units (except `Flag__User` since it could be more difficult).
- On the second condition, I separate it into two statements and assign `Min_CellWidth` to `r * CCSN_MaxRefine_RadFac`. Hope this will make it more clear.

Reply:
Good suggestion about supporting `OPT__FLAG_UNIT`